pkg/dataobj/dataobj.go

```diff
@@ -286,15 +327,10 @@ func streamSizeEstimate(stream logproto.Stream) int {
 // If Flush builds an object but fails to upload it to object storage, the
 // built object is cached and can be retried. [Builder.Reset] can be called to
 // discard any pending data and allow new data to be appended.
-func (b *Builder) Flush(ctx context.Context) error {
-	switch b.state {
-	case builderStateEmpty:
-		return nil // Nothing to flush
-	case builderStateDirty:
-		if err := b.buildObject(); err != nil {
-			return fmt.Errorf("building object: %w", err)
-		}
-		b.state = builderStateFlush
+func (b *Builder) Flush(ctx context.Context) (string, error) {
+	_, err := b.FlushToBuffer()
```

Comment:
By the way, this changes the behaviour of Flush where calling Flush immediately after a successful flush will cause it to re-write the same object.

Reply:
I've moved Reset back into this method now I'm returning the FlushResult summary. Does that fix the issue?